script.js

```diff
@@ -7,54 +7,44 @@ const recipeModalCloseBtn = document.getElementById("recipe-modal-close-btn")
 const recipeModal = document.getElementById("recipe-modal")
 const recipeInnerModal = document.getElementById("recipe-modal-inner")
 
-
 //Close the modal button 
+function closeModal() {
+    recipeModal.style.display = 'none'
+}
+
 recipeModalCloseBtn.addEventListener("click", closeModal)
+
 document.addEventListener("keydown", function (e) {
     if (e.key === "Escape") {
         closeModal()
     }
 })
 
-function closeModal() {
-    recipeModal.style.display = 'none'
-}
-
-//Open the modal
-
-getRecipeBtn.addEventListener("click", renderRecipe)
-
 function getMatchingRecipesArray() {
-
     //selectedCuisine is the one with radio input selected
-    if (document.querySelector('input[type="radio"]:checked')) {
-        const selectedCuisine = document.querySelector('input[type="radio"]:checked').value
-
-
+    const selectedCuisine = document.querySelector('input[type="radio"]:checked')?.value
+    if (selectedCuisine) {
```

Comment:
To prevent running `querySelector` twice, we can add a `?` before the `.value`. If the element cannot be found, `selectedCuisine` will be `undefined`, otherwise we'll set it to the element's `value`.

If we _don't_ use the `?`, we'll get an error when trying to access `.value` if the element can't be found.

We then need to check for `undefined` by using `if (selectedCuisine)`